internal/pkg/service/common/dependencies/public.go

```diff
@@ -21,32 +21,61 @@ type public struct {
 	encryptionAPIClient client.Client
 	stackFeatures       storageapi.FeaturesMap
 	stackServices       storageapi.ServicesMap
-	components          *model.ComponentsProvider
+	components          Lazy[*model.ComponentsProvider]
 }
 
-func NewPublicDeps(ctx context.Context, base Base, storageAPIHost string) (v Public, err error) {
+type PublicDepsOption func(*publicDepsConfig)
+
+type publicDepsConfig struct {
+	includeComponents bool
+}
+
+// WithComponents defines if the components should be retrieved from Storage index.
```

Comment:
Tento komentar by sa mohol mozno este vylepsit, `be retrieved from Storage index ON STARTUP`/`ON INIT`?